arangod/Agency/State.cpp

```diff
@@ -293,10 +293,21 @@ index_t State::logNonBlocking(index_t idx, velocypack::Slice const& slice,
                               term_t term, uint64_t millis, std::string const& clientId,
                               bool leading, bool reconfiguration) {
   _logLock.assertLockedByCurrentThread();
-
-  auto byteSize = slice.byteSize();
-  auto buf = std::make_shared<Buffer<uint8_t>>();
-  buf->append((char const*)slice.begin(), byteSize);
+  
```

Comment:
As far as I see this is - contrary to the comment above it - called for both the leader and the follower log appending. This means we would log it 3 times, if we do the logging here. There is something to be said for this, since when the leader appends something to its log, we don't know if it will stay in the Raft log, before it is replicated to a majority of followers. On the other hand, it uses three times as much disk space as needed.
Furthermore, I think at this stage the preconditions are already fulfilled and stripped. In some circumstances it could be useful to also log failed attempts to write something to the raft log. For me it is unclear what the best way to proceed here is.

Reply:
As far as I can see setting the log level to `DEBUG` will make it log only on the leader, but not on followers. At least I am checking the `leading` flag here, and if it is false (i.e. we are on a follower), we log only if log level is `TRACE`.
That means one should be able to run it _only_ on leaders, if so desired.

But I am not an expert here, and if there is a better place to do the logging, I would be happy to know it!